faucet.go

```diff
@@ -948,21 +893,33 @@ func (l *lightningFaucet) payInvoice(homeTemplate *template.Template,
 	homeState *homePageContext, w http.ResponseWriter, r *http.Request) {
 
 	// Disable pay invoice if user set this parameter
-	if l.disablePayInvoices {
+	if l.cfg.DisablePayInvoices {
 		http.Error(w, "invoices payment was disabled", 403)
 		return
 	}
 
-	elapsed := time.Since(lastPayInvoiceTime)
-	if elapsed < PayInvoiceTimeout {
-		homeState.SubmissionError = PayInvoiceTimeNotElapsed
+	// Get the invoice from users form and set the input value again
+	// for the users repeat the action or verify
+	rawPayReq := r.FormValue("payinvoice")
+	homeState.FormFields["Payinvoice"] = rawPayReq
+
+	// Verify IP before open a channel
+	hostIP, err := getRealIP(r, l.cfg.UseRealIP)
```

Comment:
You may consider validating `hostIP`.  But the only way it would not be a valid IP address if the upstream proxy is malfunctioning or malicious.

Reply:
I prefer to follow the actual implementation and explain better the usage of a reverse proxy.